src/hal/i_components/knob2float.icomp

```diff
@@ -34,8 +34,8 @@ license "GPL";
 
 FUNCTION(_)
 {
-    long delta_counts;
-    double tmp_out;
+    hal_float_t delta_counts;
```

Comment:
No, take a look at the input pin it is s32 probably some encoder input. Therefore, old_count and delta_count should be s32 too. The conversion happens at line 45 where the delta_counts are scaled and added to the float value.